### PR TITLE
ci: upgrade gh actions checkout and setup-node to v4

### LINKED ISSUE #8167: CI `uses node12 which is deprecated`

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #8167.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -9,9 +9,9 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - name: Setup Node.js 18.x
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v4
```
Comment (hamirmahal) [inline .github/workflows/test.yml:14]: These are the latest major versions as of the time of this writing.

**Review by DhruvBundheliya — COMMENTED** [review]
+1

**Review by ad1992 — APPROVED** [review]
Thanks for the PR @hamirmahal

**Comment by hamirmahal** [comment]
You're welcome @ad1992.

**Pull request merged**